### PR TITLE
Hdh306 migrate gitness docs

### PR DESCRIPTION
**Pull request by dewan-ahmed**
## Description

* Please describe your changes: This PR brings the files from https://docs.gitness.com/ and adds them under docs/open-source. 
* Jira/GitHub Issue numbers (if any): https://harness.atlassian.net/browse/HDH-306
* Preview links/images (Internal contributors only): _to be added_

Please ensure we review the following:

- [ ] src/components files are properly defined for Harness Open Source.
- [ ] If any CSS change is needed for this work.
- [ ] All required images/logos/icons for Harness Open Source are added.
- [ ] All the internal links work after the migration. For example, any `/docs` link from Gitness Docs will be `/docs/open-source` after this migration.
- [ ] IMPORTANT: We're not breaking anything during this migration.

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

**Comment by dewan-ahmed** [comment]
Known Issues:

1. Navigating to /docs/open-source keeps opening up /docs/docs/open-source.../docs/docs/docs//open-source
Screenshot 2024-09-18 at 1 03 55 PM

2. I copied the template from Code Repo so I'll update few of these links. Since there's no Release Notes or Knowledge Base for Harness Open Source, we'll replace these with something else, like "Getting Started" and "Tutorials".

Screenshot 2024-09-18 at 1 05 36 PM

**Comment by SushrutHarness** [comment]
I've created a PR into this one to fix the broken links: https://github.com/dewan-ahmed/developer-hub/pull/1

Plus I just merged a commit fixing known issue 1 above.

**Comment by rohanmaharjan100** [comment]
@dewan-ahmed  Can we temporarily use the Gitness logo on the sidebar until the new module icon for Open Source is ready?
image
Logo on side bar can be added as :

- https://github.com/harness/developer-hub/blob/main/static/img/icon_gitness.svg
- https://github.com/rohanmaharjan100/developer-hub/blob/c28a23d59729647b23530cd7bbd01b6a2a4a0c70/src/css/custom.css#L1264

File: src/components/Docs/OpenSource.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import { docsCards } from "./data/openSourceData";
+
+export default function Code() {
+  const { siteConfig: { baseUrl = "/" } = {} } = useDocusaurusContext();
+  return (
+    <div className="container">
+      <div className={styles.topSection}>
+        <div className={styles.spaceBetween}>
+          <div className={styles.moduleTitle}>
+            <img src={`${baseUrl}img/icon_code.svg`} />
```
Comment (rohanmaharjan100) [inline src/components/Docs/OpenSource.tsx:17]: Using Code Repository icon here ?

**Comment by dewan-ahmed** [comment]
> @dewan-ahmed Can we temporarily use the Gitness logo on the sidebar until the new module icon for Open Source is ready? image Logo on side bar can be added as :
> 
> * https://github.com/harness/developer-hub/blob/main/static/img/icon_gitness.svg
> * https://github.com/rohanmaharjan100/developer-hub/blob/c28a23d59729647b23530cd7bbd01b6a2a4a0c70/src/css/custom.css#L1264

hoss_logo_reverse
icon_opensource
@rohanmaharjan100 we have the logos

File: src/components/Docs/OpenSource.tsx (hunk 1 above)
Comment (dewan-ahmed) [reply src/components/Docs/OpenSource.tsx:17]: Used Code Repo as template. Some of the lines need to be updated to reference Harness Open SOurce.

**Comment by dewan-ahmed** [comment]
Release Notes link for Harness Open Source can point to https://github.com/harness/gitness/releases @rohanmaharjan100

**Comment by SushrutHarness** [comment]
Update Whats Support to API Docs reference link

**Comment by dewan-ahmed** [comment]
@rohanmaharjan100 for the icon, we can use Harness icon as Harness Open Source icon.

**Comment by dewan-ahmed** [comment]
@SushrutHarness https://github.com/harness/developer-hub/pull/8045/commits/dc00264cb54addc90b8d77f83947f9a665487d3e renames applicable 'Gitness' texts with 'Harness Open Source'. There are still many references like the environment variables which we cannot change until those are updated within product first. This commit also touches some CD pages. Please review 👀  Thanks.

**Review by pratmit — APPROVED** [review]
LGTM

**Comment by dewan-ahmed** [comment]
I aim to merge this PR around 11 AM PST. Let's not add anymore commits unless something major is broken. We can continue updating post-merge.

**Review by SushrutHarness — APPROVED** [review]
LGTM

**Pull request merged**